drivers/system_pathways/app/models/system_pathways/chart_base.rb

```diff
@@ -0,0 +1,447 @@
+      scope = clients
+      scope = filter_for_race(scope)
+      scope = filter_for_ethnicity(scope)
+      scope = filter_for_veteran_status(scope)
+      scope = filter_for_ce_involvement(scope)
+      scope = filter_for_head_of_household(scope)
+      scope = filter_for_chronic_at_entry(scope)
+      scope = filter_for_disabling_condition(scope)
+      scope
+    end
+
+    def client_word
+      return 'Households' if filter.hoh_only
+
+      'Clients'
+    end
+
+    def sanitized_node(node)
+      return node if node.in?(['Returns to Homelessness', 'Served by Homeless System'])
+
+      available = report.class.available_project_types.map do |p_type|
+        HudUtility.project_type_brief(p_type)
+      end + destination_lookup.keys + ph_projects.values
+
+      available.detect { |m| m == node }
+    end
+
+    def destination_lookup
+      {
+        'Permanent Destinations' => 'destination_permanent',
+        'Homeless Destinations' => 'destination_homeless',
+        'Institutional Destinations' => 'destination_institutional',
+        'Temporary Destinations' => 'destination_temporary',
+        'Other Destinations' => 'destination_other',
+      }
+    end
+
+    private def ph_projects
+      [
+        'PH - PSH',
+        'PH - PH',
+        'PH - OPH',
+        'PH - RRH',
+      ].map { |m| [m, "#{m} Pre-Move in"] }.to_h
+    end
+
+    def average(value, count)
+      return 0 unless count.positive?
+
+      value.to_f / count
+    end
+
+    private def filter_for_disabling_condition(scope)
+      return scope if @filter.disabling_condition.nil?
+
+      scope.merge(SystemPathways::Enrollment.where(disabling_condition: filter.disabling_condition))
+    end
+
+    private def filter_for_chronic_at_entry(scope)
+      return scope if @filter.chronic_status.nil?
+
+      scope.merge(SystemPathways::Enrollment.where(chronic_at_entry: filter.chronic_status))
+    end
+
+    private def filter_for_race(scope)
+      return scope unless filter.races.present?
+
+      race_scope = nil
+      filter.races.each do |column|
+        next if column == 'MultiRacial'
+
+        race_scope = add_alternative(race_scope, SystemPathways::Client.where(column.underscore.to_sym => true))
+      end
+
+      # Include anyone who has more than one race listed, anded with any previous alternatives
+      race_scope ||= scope
+      race_scope = race_scope.where(id: multi_racial_clients.select(:id)) if filter.races.include?('MultiRacial')
+      scope.merge(race_scope)
+    end
+
+    private def multi_racial_clients
+      # Looking at all races with responses of 1, where we have a sum > 1
+      a_t = SystemPathways::Client.arel_table
+      columns = [
+        :am_ind_ak_native,
+        :asian,
+        :black_af_american,
+        :native_hi_pacific,
+        :white,
+      ].map do |col|
+        "CASE WHEN #{a_t[col].to_sql} THEN 1 ELSE 0 END"
+      end
+      scope.where(Arel.sql(columns.join(' + ')).between(2..98))
```

Comment:
Since the cases will only return 0 or 1, 99 can't occur:
```suggestion
      scope.where(Arel.sql(columns.join(' + ')).gt(1))
```